clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3948,6 +3964,20 @@ protected boolean shouldUpdateUpstreamOffset(PubSubMessage<KafkaKey, KafkaMessag
    */
   protected abstract Set<String> maybeSendIngestionHeartbeat();
 
+  private void mayResumeRecordLevelMetricsForCurrentVersion() {
+    if (recordLevelMetricEnabled.get()) {
+      return;
+    }
+    for (PartitionConsumptionState partitionConsumptionState: partitionConsumptionStateMap.values()) {
```

Comment:
This is consuming, so you are resuming after ready-to-serve for current version?

tiny nit: may->maybe as

Reply:
Yes, and the metrics are still useful at the steady state.